src/main/java/org/cryptomator/ui/fxapp/FxApplication.java

```diff
@@ -75,6 +76,23 @@ public void start() {
 			appWindows.checkAndShowUpdateReminderWindow();
 		}
 
+		var dokany = "org.cryptomator.frontend.dokany.mount.DokanyMountProvider";
+		boolean dokanyInfoWindowShown = false;
+		if (settings.mountService.getValueSafe().equals(dokany)) {
+			appWindows.showDokanyInfoWindow();
+			dokanyInfoWindowShown = true;
+			settings.mountService.set(null);
+		}
+		for (VaultSettings vaultSettings : settings.directories) {
+			if (vaultSettings.mountService.getValueSafe().equals(dokany)) {
+				if (!dokanyInfoWindowShown) {
+					appWindows.showDokanyInfoWindow();
+					dokanyInfoWindowShown = true;
+				}
+				vaultSettings.mountService.set(null);
+			}
+		}
```

Comment:
To increase the readability and linearize the logic i suggest to first search & replace and then show the dialog. We can also stream the vault directories:

```
                boolean dokanyFound = false;
                //check general settings
		if (settings.mountService.getValueSafe().equals(dokany)) {
			dokanyFound = true;
			settings.mountService.set(null);
		}
		//check each vault setting
		for( vaultSettings: settings.directories) {
		  if(/*vaultSettings contain dokanyProviderId*/)  {
		    dokanyFound = true;
		    vaultSettings.mountService.set(null);
		  }
		}
		
		//show dialog
		if (dokanyFound) /* showDialog */
```

Also i suggest to refactor it into its own Method